src/safe_cli/tx_builder/tx_builder_file_decoder.py

```diff
@@ -0,0 +1,240 @@
+
+
+def _get_base_field_type(field_type: str) -> str:
+    trimmed_value = field_type.strip()
+    if not trimmed_value:
+        raise SoliditySyntaxError("Empty base field type for")
+
+    base_field_type_regex = re.compile(
+        r"^([a-zA-Z0-9]*)(((\[\])|(\[[1-9]+[0-9]*\]))*)?$"
+    )
+    match = base_field_type_regex.match(trimmed_value)
+    if not match:
+        raise SoliditySyntaxError(f"Unknown base field type from {trimmed_value}")
+    return match.group(1)
+
+
+def _is_array(values: str) -> bool:
+    trimmed_value = values.strip()
+    return trimmed_value.startswith("[") and trimmed_value.endswith("]")
+
+
+def parse_array_of_values(values: str, field_type: str) -> List[Any]:
+    if not _is_array(values):
+        raise SoliditySyntaxError("Invalid Array value")
+
+    parsed_values = parse_string_to_array(values)
+    return [
+        (
+            parse_array_of_values(item_value, field_type)
+            if _is_array(item_value)
+            else parse_input_value(_get_base_field_type(field_type), item_value)
+        )
+        for item_value in parsed_values
+    ]
+
+
+def is_boolean_field_type(field_type: str) -> bool:
+    return field_type == "bool"
+
+
+def is_int_field_type(field_type: str) -> bool:
+    return field_type.startswith("uint") or field_type.startswith("int")
+
+
+def is_tuple_field_type(field_type: str) -> bool:
+    return field_type.startswith("tuple")
+
+
+def is_bytes_field_type(field_type: str) -> bool:
+    return field_type.startswith("bytes")
+
+
+def is_array_of_strings_field_type(field_type: str) -> bool:
+    return field_type.startswith("string[")
+
+
+def is_array_field_type(field_type: str) -> bool:
+    pattern = re.compile(r"\[\d*\]$")
+    return bool(pattern.search(field_type))
+
+
+def is_multi_dimensional_array_field_type(field_type: str) -> bool:
+    return field_type.count("[") > 1
+
+
+def parse_input_value(field_type: str, value: str) -> Any:
+    trimmed_value = value.strip() if isinstance(value, str) else value
+
+    if is_tuple_field_type(field_type):
+        return tuple(json.loads(trimmed_value))
+
+    if is_array_of_strings_field_type(field_type):
+        return json.loads(trimmed_value)
+
+    if is_array_field_type(field_type) or is_multi_dimensional_array_field_type(
+        field_type
+    ):
+        return parse_array_of_values(trimmed_value, field_type)
+
+    if is_boolean_field_type(field_type):
+        return parse_boolean_value(trimmed_value)
+
+    if is_int_field_type(field_type):
+        return parse_int_value(trimmed_value)
+
+    if is_bytes_field_type(field_type):
+        return HexBytes(trimmed_value)
+
+    return trimmed_value
+
+
+@dataclasses.dataclass
+class SafeProposedTx:
+    id: int
+    to: str
+    value: int
+    data: str
+
+    def __str__(self):
+        return f"id={self.id} to={self.to} value={self.value} data={self.data}"
+
+
+def convert_to_proposed_transactions(
+    batch_file: Dict[str, Any]
+) -> List[SafeProposedTx]:
```

Comment:
Is this new functionality? If so I suggest moving it into a different Pull Request. The scope of this PR is already quite big to introduce new functionality with it in my opinion (and reverting the CLI changes proposed here should not necessarily revert this new feature).

Reply:
I have finally reverted the changes related to tx-builder. I have created a branch from this one incorporating them. The PR of the new branch is: https://github.com/safe-global/safe-cli/pull/426